### PR TITLE
Add photo-demo.lua to take snapshots at various graphics quality settings

### PR DESCRIPTION
**Pull request by nat-goodspeed**
The other commits are support features used by photo-demo.lua.

### CODE REVIEW

File: indra/newview/lluilistener.cpp [hunk 1]
```diff
@@ -443,6 +448,11 @@ void LLUIListener::uploadLocalTexture(const LLSD& event) const
     });
 }
 
+LLSD LLUIListener::getGraphicsQuality(LLSD const& request)
```
Comment (maxim-productengine) [inline indra/newview/lluilistener.cpp:451]: We can use `LLDebugSettings.get('RenderQualityPerformance')` in Lua

**Review by maxim-productengine — APPROVED** [review]

File: indra/newview/lluilistener.cpp (hunk 1 above)
Comment (nat-goodspeed) [reply indra/newview/lluilistener.cpp:451]: You're right. It would be cleaner to revert the change I added.

Comment (nat-goodspeed) [reply indra/newview/lluilistener.cpp]: https://github.com/secondlife/viewer/pull/3268/commits/ca5b1e4ea7aa8cb012356c44133fa5037d202449

Comment (maxim-productengine) [reply indra/newview/lluilistener.cpp]: ship it

**Pull request merged**